Demos/DynamicLoader/DynExec/main.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019-2020, Andrew Kaster <andrewdkaster@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <stdio.h>
+
+int libfunc();
+
+extern "C" void __libc_init();
+
+int main(int argc, char** argv)
+{
+    (void)argc;
+    (void)argv;
+    libc_testfunc();
+    // __libc_init();
```

Comment:
Instead of trying to use libc right out of the gate, it might make sense to see if we can get just an PIE running via the  loader that doesn't have any library dependencies first. 

Maybe a hacky manual dbgputstr "Hello World"? Without worrying about DT_NEEDED until this part works.

Something like
```c++
#include <Kernel/syscall.h>
const char* g_string = "Hello, World!\n";

void local_dbgputstr(const char* str, int len)
{
    constexpr unsigned int function = Kernel::SC_dbgputc;
    for (int i = 0; i < len; ++i) {
        unsigned int result;
        asm volatile("int $0x82"
                 : "=a"(result)
                 : "a"(function), "d"((u32)str[i])
                 : "memory");
    }
}

int main (int argc, char** argv)
{
    local_debugputstr(g_string, 15);
    return 0;
}
```